scripts/test-crds.sh

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env bash
+
+# ------------------------------------------------------------------------------
+# test-crds.sh
+#
+# This script tests that our legacy v1beta1 versions of our CRDs upgrade cleanly
+# to their v1 versions, and that both the legacy and the v1 versions of the CRDs
+# both function with our controllers.
+#
+# Test Workflow
+#
+#   Step 1
+#
+#   1. deploy the legacy CRDs
+#   2. install the chart
+#   3. verify legacy CRDs are in place
+#   4. run helm tests (via legacy CRDs)
+#
+#   Step 2
+#
+#   1. cleanup the chart
+#   2. deploy the v1 CRDs
+#   3. verify the CRDs updates
+#   4. redeploy the chart
+#   5. run helm tests (via v1 CRDs)
+#
+# NOTE: these tests are sensitive to which Kubernetes version they are run on.
+#       they will only run on Kubernetes clusters between v1.16.x and 1.21.x,
+#       as v1.15.x is too old for the new CRDs and v1.22.x is to new for the
+#       old CRDs. At some point in the future (potentially when <v1.22.x is no
+#       longer supported by the chart) we can remove these tests.
```

Comment:
please be more specific: "this test requires k8s between 1.16 and 1.21 inclusively because it requires support for both v1beta1 and v1 CRD apis"

Reply:
```suggestion
# NOTE: these tests are sensitive to which Kubernetes version they are run on.
#       they will only run on Kubernetes clusters between v1.16.x and 1.21.x
#       (inclusively) as v1.15.x is too old for the new CRDs and v1.22.x is to new
#       for the old CRDs. At some point in the future (potentially when <v1.22.x is
#       no longer supported by the chart) we can remove these tests.
```